### PR TITLE
Text generation logics calculation fix. 

### PR DESCRIPTION
**Pull request by yidong72**
# What does this PR do?
Only update the logits if the current token passes the context length.

### CODE REVIEW

**Review by michalivne — APPROVED** [review]
LGTM! see minor comments

File: nemo/collections/nlp/modules/common/text_generation_utils.py [hunk 1]
```diff
@@ -621,17 +630,19 @@ def sample_sequence_batch(
                 # make sure it won't sample outside the vocab_size range
                 logits[:, tokenizer.vocab_size :] = -float('Inf')
 
+                started = context_lengths <= context_length
```
[the remark this answers is not in the document]
Comment (yidong72) [reply nemo/collections/nlp/modules/common/text_generation_utils.py:642]: done

**Review by michalivne — APPROVED** [review]
LGTM!

**Comment by markelsanz14** [comment]
Gerald and I tested this. The fix works well and fixed the -inf issue we were having. LGTM
